### ISSUE TITLE
api_feedreader and TTRSS not working?

### ISSUE THREAD

**Issue by cdorin93** [jangernert/FeedReader#760]
Hey guys, 

recently got my VPS and decided to give TTRSS and feedreader a try.

TTRSS is working fine, however, I got some issues while trying to connect feedreader to it. I keep getting TTRSS API error whenever a refresh on my news feed is done. The articles are not getting retrieved... just the folders and titles of the feeds:

```
feedreader
feedreader-INFO: 02:11:20.320: FeedReader 2.2.1
(feedreader:4759): Gtk-DEBUG: 02:11:20.395: Connecting to session manager
feedreader-INFO: 02:11:20.428: Plugin Name: Tiny Tiny RSS
feedreader-INFO: 02:11:20.428: Plugin Version: 0.1
feedreader-INFO: 02:11:20.428: Plugin Website: http://jangernert.github.io/FeedReader/
feedreader-INFO: 02:11:20.428: Plugin Dir: /usr/lib64/feedreader/plugins/
feedreader-INFO: 02:11:20.554: TTRSS Session ID: g6cc7l285pu22dbsphlk65e017
feedreader-INFO: 02:11:20.554: TTRSS API Level: 14

(feedreader:4759): feedreader-CRITICAL **: 02:11:20.679: ttrss api error

(feedreader:4759): feedreader-CRITICAL **: 02:11:20.680: ttrss api error

(feedreader:4759): feedreader-CRITICAL **: 02:11:20.965: FavIcon.load: [Unknown] - Unrecognized image file format
```

Checked also the TTRSS logs - guess here is the issue?. Got the following error:

```
Error: E_USER_NOTICE (1024)	
Filename:classes/db.php:22 

Message: Legacy connect requested to mysql1. classes/db.php(22): user_error(Legacy connect requested to mysql, 1024)
2. classes/db.php(100): legacy_connect()
3. include/db.php(4): get()
4. plugins/api_feedreader/init.php(37): db_escape_string()
5. classes/api.php(505): removeLabel()
6. api/index.php(82): index(removelabel)
```

Does anyone else have this problem?  Any advice on how shall I debug this?

Have an Ubuntu 16.04 machine, apache & mariadb. Tested with Feedreader 2.2.1 (Solus repository) and Feedread 2.3.0 (flatpak), while TTRSS was installed using the master branch of their repository a couple of days ago. The PDO module is also active.

Thanks for your help! ^^

_______________

**Update (Nov 4, 2018)**

Ok, reinstalled everything now. Tried again to install api_feedreader and run feedreader.

It seems that feedreader is running (getting the news from my ttrss server) even though, when running it from terminal, I get the "ttrss api error" message (though... I am happy that my newsfeed gets updated).

The only lil weird problem is the following one:
Whenever I open for the first time feedreader (after turning my pc on for example), I get the following log message/error in my TTRSS installation: 

```
Error: E_USER_NOTICE (1024)	
Filename:classes/db.php:22 

Message: Legacy connect requested to mysql1. classes/db.php(22): user_error(Legacy connect requested to mysql, 1024)
2. classes/db.php(100): legacy_connect()
3. include/db.php(4): get()
4. plugins/api_feedreader/init.php(37): db_escape_string()
5. classes/api.php(505): removeLabel()
6. api/index.php(82): index(removelabel)
```

However, feedreader works and gets the news despite of the error mentioned above. Once feedreader runs, whenever i click on the update button, no log message/error appears in TTRSS.

**Comment by jangernert** [comment]
There was recently a PR that fixed a similar issue.
Do you by any chance have a version of the plugin that was downloaded before this was merged?
#751

**Comment by cdorin93** [comment]
Hey @jangernert . Nop, everything is up-to-date (used git clone and copy your api_feedreader into ttrss/plugins). 

I also saw the respective commit and thought feedreader is now working with TTRSS. Not sure why it is not :(.

**Comment by brendanlong** [comment]
I saw this same error today and it should be fixed in v2.8.0. The problem was that we were using a mix of PDO and legacy DB stuff for string escaping. I switched it so we exclusively use PDO now.

See bdeab5c01146320e41f4840d79108b5e245c4426

Can you upgrade the plugin and see if it works for you now?
